Packs/CircleCI/Integrations/CircleCI/CircleCI.py

```diff
@@ -0,0 +1,302 @@
+
+''' CONSTANTS '''
+DEFAULT_VCS_TYPE = 'github'
+DEFAULT_LIMIT_VALUE = 20
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+
+    def __init__(self, base_url: str, api_key: str, verify: bool, proxy: bool, vc_type: str, organization: str,
+                 project: str):
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, headers={'authorization': f'Basic {api_key}'})
+        self.vc_type = vc_type
+        self.organization = organization
+        self.project = project
+
+    def get_job_artifacts(self, vc_type: str, organization: str, project: str, job_name: str):
+        return self._http_request(
+            method='GET',
+            url_suffix=f'/project/{vc_type}/{organization}/{project}/{job_name}/artifacts'
+        )
+
+    def get_workflows_list(self, vc_type: str, organization: str, project: str, page_token: Optional[str] = None):
+        url_suffix = f'insights/{vc_type}/{organization}/{project}/workflows'
+        return self._http_request(
+            method='GET',
+            url_suffix=url_suffix,
+            params={'page-token': page_token} if page_token else None
+        )
+
+    def get_last_workflow_runs(self, vc_type: str, organization: str, project: str, workflow_name: str,
+                               page_token: Optional[str] = None):
+        url_suffix = f'insights/{vc_type}/{organization}/{project}/workflows/{workflow_name}'
+        return self._http_request(
+            method='GET',
+            url_suffix=url_suffix,
+            params={'page-token': page_token} if page_token else None
+        )
+
+    def get_workflow_jobs(self, workflow_id: str, page_token: Optional[str] = None):
+        url_suffix = f'/workflow/{workflow_id}/job'
+        return self._http_request(
+            method='GET',
+            url_suffix=url_suffix,
+            params={'page-token': page_token} if page_token else None
+        )
+
+
+''' HELPER FUNCTIONS '''
+
+
+def get_response_with_pagination(client_command: Callable, client_command_args: List, limit: int) -> List[Dict]:
+    """
+    Preforms API calls to CircleCI, using pagination mechanism given by CircleCI.
+    CircleCI gives a page token for retrieving next page if more results exists.
+    Args:
+        client_command (Callable): Client function command which performs the API call.
+        client_command_args (Dict): Args for client command.
+        limit (int): Maximum number of results to retrieve.
+
+    Returns:
+        (List[Dict]): List of the results.
+    """
+    results: List[Dict] = []
+    response = client_command(*client_command_args)
+    results.extend(response.get('items', []))
+    while len(results) < limit:
+        next_page_token: Optional[str] = response.get('next_page_token')
+        if not next_page_token:
+            break
+        response = client_command(*(client_command_args + [next_page_token]), next_page_token=next_page_token)
```

Comment:
is the `next_page_token=next_page_token` required here?

Reply:
Good question. probably not. I'll re-check this